### PR TITLE
refactor(select): wait for interactions before opening modal

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [] 

### Description

Related to these two issues with the <Select /> component that surfaced from Sprintly QA:

[case 1](https://www.notion.so/artsy/Not-able-to-max-bid-on-artwork-buggy-select-dropdowns-on-iphone-13-mini-video-included-53053dc615644192a1e7af1a6111a4d8?pvs=4)

[case 2](https://www.notion.so/artsy/Can-t-register-and-bid-on-a-live-auction-2c48b52a7cd9400db40dcd1a400dc5cf?pvs=4)

We weren't able to reproduce (not even on my device that I originally recorded the videos on) after reopening the app.

But I believe that it would be better to use [InteractionsManager](https://reactnative.dev/docs/interactionmanager) in order to prevent this from happening again.

Happy to close it though if anyone feels strongly that this will **not** add any value.

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- refactor(select): wait for interactions before opening modal - gkartalis

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -1312,4 +1312,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: d7802ab816bd45141783671dcaf393f2a878fd83
 
-COCOAPODS: 1.11.3
+COCOAPODS: 1.12.1
```
Comment (gkartalis) [inline ios/Podfile.lock:1315]:
```suggestion
COCOAPODS: 1.11.3
```

**Review by MounirDhahri — APPROVED** [review]
🙌

**Comment by MounirDhahri** [comment]
the changelog looks awkward but it's alright 😄

**Comment by MounirDhahri** [comment]
weird bug 😄

File: src/app/Components/Select/Select.tsx [hunk 2]
```diff
@@ -78,7 +78,10 @@ export const Select = <ValueType,>({
 
   const open = async () => {
     await blurAllOtherInputs()
-    setShowingModal(true)
+
+    InteractionManager.runAfterInteractions(() => {
+      setShowingModal(true)
+    })
```
Comment (damassi) [inline src/app/Components/Select/Select.tsx:84]: Good to know 👍

**Pull request merged**